data/grub_auth/create_custom_grub.sh

```diff
@@ -3,47 +3,62 @@
 set -euo pipefail
 
 ROOT_DISK=`df -T | grep \/$ | awk '{print $1}'`
-PARTNO=`echo ${ROOT_DISK:0-1}`
 FS=`df -T | grep \/$ | awk '{print $2}'`
+UUID=`blkid | grep $ROOT_DISK | awk -F \" '{print $2}'`
+ARCH=`uname -m`
+KERNEL=""
 SUP_USER="$1"
 SUP_PASSWD="$2"
 MAINT_USER="$3"
 MAINT_PASSWD="$4"
+SUPER_PDKDF2=`cat /tmp/sup_passwd_hash | grep PBKDF2 | awk -F "password is " '{print \$2}'`
+MAINT_PDKDF2=`cat /tmp/maint_passwd_hash | grep PBKDF2 | awk -F "password is " '{print \$2}'`
 
 : > /etc/grub.d/10_linux
 
 if [ $FS = "ext4" ] || [ $FS = "ext3" ]; then
 FS=ext2
 fi
 
+if [ $ARCH = "x86_64" ]; then
+KERNEL="vmlinuz"
+elif [ $ARCH = "aarch64" ]; then
+KERNEL="Image"
+elif [ $ARCH = "s390x" ]; then
```

Comment:
Just curious if aarch64 and s390x can put them in the same elif (Dont know if it works :)

Reply:
Good comments there, I did try the script on all platforms. it worked fine.
But for ppc64le and s390x, we need user ssh serial console rather than needle to match the boot entry. (will implement it in phase 2)

Reply:
okay , thanks for comments